lintlizard/__init__.py

```diff
@@ -34,38 +35,72 @@ def version_command(self) -> Command:
 
 TOOLS = [
     CommandTool('flake8'),
-    CommandTool('isort', run_params=('-c', '.'), fix_params=('.',)),
-    CommandTool('mypy'),
-    CommandTool('black', run_params=('--check', '.'), fix_params=('.',)),
+    CommandTool('isort', run_params=('-c',), fix_params=tuple()),
+    CommandTool('mypy', allow_specify_files=False),
+    CommandTool('black', run_params=('--check',), fix_params=tuple()),
 ]
 
 
-def execute_tools(fix: bool) -> Iterable[bool]:
+def execute_tools(fix: bool, files: Tuple[str, ...]) -> Iterable[bool]:
     tools = TOOLS
     if fix:
         tools = [tool for tool in tools if tool.fixable]
     for tool in tools:
         print('*' * 79)
         try:
             subprocess.run(args=tool.version_command(), check=True)
-            cmd = tool.fix_command() if fix else tool.run_command()
-            subprocess.run(args=cmd, check=True)
+            cmd = (
+                tool.fix_command()
+                if fix and tool.fix_params is not None
+                else tool.run_command()
+            )
+            subprocess.run(args=cmd + files, check=True)
```

Comment:
With `mypy`, you often don't want to type-check everything. Instead, you specify what to check in its config files. Usually it's just production code, not any tests or supplementary stuff, so if the tool does _not_ `allow_specify_files` then we shouldn't give it anything at all, regardless of what's passed in/changed/default `.`.